subsys/bluetooth/services/ots/ots.c

```diff
@@ -187,14 +187,14 @@ ssize_t ots_obj_name_write(struct bt_conn *conn,
 		rc = bt_gatt_ots_obj_manager_next_obj_get(ots->obj_manager, obj, &obj);
 	}
 
-	/* Update real object name after no duplicate detected */
-	strcpy(ots->cur_obj->metadata.name, name);
-
+	/* No duplicate detected, notify application and update real object name */
 	if (ots->cb->obj_name_written) {
 		ots->cb->obj_name_written(ots, conn, ots->cur_obj->id,
-					  ots->cur_obj->metadata.name);
+					  ots->cur_obj->metadata.name, name);
 	}
 
+	strcpy(ots->cur_obj->metadata.name, name);
```

Comment:
I recently added a new function to Zephyr, `utf8_lcpy`, which does a properly truncated cpy of UTF8 strings (which OTS supports). We should update OTS to use that when copying the UTF8-encoded names.

Reply:
I will follow up with a separate PR for this one.